text/0000-no-more-extern-crate.md

```diff
@@ -70,29 +85,21 @@ For example, `cargo build`-ing a crate `my_crate` with a dependency on `rand`
 results in a call to rustc that looks something like
 `rustc --crate-name mycrate src/main.rs --extern rand=/path/to/librand.rlib ...`.
 
-When an external crate is specified this way,
-the crate will automatically brought into scope as if an
-`extern crate name_of_crate;`
-declaration had been added to the current crate root.
-This behavior won't occur when including a library using the `-l`
+When an external crate is specified this way, it will be automatically
+available to any module in the current crate through `use` statements or
+absolute paths (e.g. `::rand::random()`). It will _not_ be automatically
+imported at root level as happens with current `extern crate`.
```

Comment:
@petrochenkov To me, it's just a simple extension of the difference between external visibility and internal visibility. Ideally, external visibility would imply internal visibility, but to me it's much more important for code in root module to work the same as code in other modules. I haven't yet found a way to consolidate those two in backwards-compatible way, maybe you'll have more luck. Possibly the cleanest solution would be not to treat the root module as the root of the namespace, but that obviously won't fly.

Anyway, I think the confusion only exists if the explanation is not framed the right way. The explanations just have to be careful with it.